docs/data/data-grid/state/state.md

```diff
@@ -66,40 +66,39 @@ Some selectors are yet to be documented.
 
 ## Save and restore the state
 
-> ⚠️ This feature isn't implemented yet. It's coming.
+The current state of the grid can be exported using `apiRef.current.exportState()`.
+It can then be restored by either passing it to the `initialState` prop or to the `apiRef.current.restoreState()` method.
+
+Watch out for controlled models and their callbacks (`onFilterModelChange` if you use `filterModel`, for instance), as the grid will call those callbacks when restoring the state.
+But if the callback is not defined or if calling it does not update the prop value, then the restored value will not be applied.
+
+### Restore the state with `initialState`
+
+You can pass the state returned by `apiRef.current.exportState()` to the `initialState` prop.
+In the demo bellow, clicking on _Recreate the 2nd grid_ will re-mount the 2nd grid with the current state of the 1st grid.
+
+> ⚠️ If you restore the page using `initialState` before the data is fetched, the grid will automatically move to the 1st page.
+
+{{"demo": "RestoreStateInitialState.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Restore the state with `apiRef` [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+{{"demo": "RestoreStateApiRef.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
We miss a paragraph explaining how to interact with the demo.

Reply:
Done